app/src/main/java/br/com/dionataferraz/vendas/profile/ProfileActivity.kt

```diff
@@ -0,0 +1,72 @@
+package br.com.dionataferraz.vendas.profile
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.widget.ProgressBar
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import br.com.dionataferraz.vendas.HomeActivity
+import br.com.dionataferraz.vendas.databinding.ActivityProfileBinding
+import br.com.dionataferraz.vendas.login.LoginActivity
+import br.com.dionataferraz.vendas.login.LoginViewModel
+import br.com.dionataferraz.vendas.login.data.remote.LoginDataSource
+import br.com.dionataferraz.vendas.login.data.request.NewUserRequest
+
+class ProfileActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityProfileBinding
+    private lateinit var viewModel: ProfileViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        ActivityProfileBinding.inflate(layoutInflater).run {
+            binding = this
+            setContentView(root)
+        }
+
+        setContentView(binding.root)
+        viewModel = ProfileViewModel()
+
+        binding.btSave.setOnClickListener {
+            getProgressBar().bringToFront();
+            getProgressBar().visibility = View.VISIBLE
+            val newUser = NewUserRequest(
+                name = binding.etName.text.toString(),
+                email = binding.etEmail.text.toString(),
+                password = binding.etPassword.text.toString(),
+            )
+            viewModel.register(newUser)
```

Comment:
A criação da model poderia estar na viewmodel, dessa forma a view não precisaria conhecer como criar uma model de request 
```suggestion
            binding.run {
              viewModel.register(
                  name = etName.text.toString(),
                  email = etEmail.text.toString(),
                  password = etPassword.text.toString(),
              )
            }
```